### ISSUE TITLE
Permit dash and underscore in macro names

### ISSUE THREAD

**Issue by delphidabbler** [delphidabbler/vied#51]
The decision to limit macro names to alpha-numeric characters is quite limiting when using multi-word names.

I find the name `my-macro` or `my_macro` to be more readable than `mymacro` or even `MyMacro`.

So it would be good to allow one or both the dash and the underscore in macro names.

**Comment by delphidabbler** [comment]
Possible implementation that also fixes bug #52:

```pascal
class function TVInfo.IsValidMacroName(const N: string): Boolean;
var
  Ch: Char;
  Idx: Integer;
begin
  if N = '' then
    Exit(False); // fixes issue #52
  if not TCharacter.IsLetter(N[1]) then
    Exit(False);
  Result := True;
  for Idx := 2 to Length(N) do
    if not TCharacter.IsLetterOrDigit(N[Idx]) 
      and (N[Idx] <> `-`) and (N[Idx] <> `_`)  then
      Exit(False);
end;
```

**Comment by delphidabbler** [comment]
> Possible implementation that also fixes bug #52:
> 
> ```pascal
> class function TVInfo.IsValidMacroName(const N: string): Boolean;
> var
>   Ch: Char;
>   Idx: Integer;
> begin
>   if N = '' then
>     Exit(False); //! fix part of issue #52
>   if not TCharacter.IsLetter(N[1]) then
>     Exit(False); //! fix remainder of issue #52
>   Result := True;
>   for Idx := 2 to Length(N) do
>     if not TCharacter.IsLetterOrDigit(N[Idx]) 
>       and (N[Idx] <> `-`) and (N[Idx] <> `_`)  then
>       Exit(False);
> end;
> ```

This code disallows macro names beginning with digits. To do this will be a breaking change. According to the Help topic [`Source/Help/HTML/whatis-fields.htm`](https://github.com/delphidabbler/vied/blob/382fbc85fbf8efbec3658a326943dd3fc05d47e0/Source/Help/HTML/whatis-fields.htm), 

```html
<p>
  Macro names are restricted to alphabetic and numeric characters. Examples
  of valid names are <code>myMacro</code>, <code>Macro1</code>,
  <code>M</code>, <code>1MyMacro</code> and <code>42</code>. Examples of
  invalid macros are <code>my-macro</code>, <code>$myMacro</code>, <code>my
  macro</code> and <code>#</code>.
</p>
```

valid macro names _can_ begin with a digit, and can in fact be made up entirely of digits.

While enabling `-` & `_` to be used within a name doesn't break backwards compatibility, disallowing leading digits certainly will. Therefore the checking code needs to change to:

```pascal
class function TVInfo.IsValidMacroName(const N: string): Boolean;
var
  Ch: Char;
  Idx: Integer;
begin
  if N = '' then
    Exit(False); //! fix part of issue #52
  if not TCharacter.IsLetterOrDigit(N[1]) then
    Exit(False); //! fix remainder of issue #52
  Result := True;
  for Idx := 2 to Length(N) do
    if not TCharacter.IsLetterOrDigit(N[Idx]) 
      and (N[Idx] <> `-`) and (N[Idx] <> `_`)  then
      Exit(False);
end;
```

**Comment by delphidabbler** [comment]
Following the change the characters permitted in macro names discussed in https://github.com/delphidabbler/vied/issues/51#issuecomment-1312657552, 
[`Source/Help/HTML/whatis-fields.htm`](https://github.com/delphidabbler/vied/blob/382fbc85fbf8efbec3658a326943dd3fc05d47e0/Source/Help/HTML/whatis-fields.htm) needs to change to something like:

```html
<p>
  Macro names must begin an alphabetic or numeric character.
  Subsequent characters must be either alphabetic, numeric
  or be a dash (minus sign) or an underscore.
  Examples of valid names are:
  <code>myMacro</code>, <code>Macro1</code>,
  <code>M</code>, <code>1macro</code>,
  <code>42</code>, <code>my_macro</code>, 
  <code>my-macro</code> and even <code>4-_</code>.
  Examples of invalid macro namess are
  <code>_M</code>, <code>-</code>, 
  <code>$myMacro</code>, <code>my&nbsp;macro</code>,
  <code>a&b</code> and <code>#</code>.
</p>
```

**Comment by delphidabbler** [comment]
Implemented at commit 6986c87e164a2349504b004ba593db0c53e324b7
